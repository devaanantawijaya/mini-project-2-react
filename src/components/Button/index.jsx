const Button = (props) => {
  const {
    title = "Button",
    logo = null,
    onClick = () => {},
    disable = false,
    p = "px-4",
    hiddenT = "",
    hiddenL = "",
    bg = "bg-blue-500", // default value untuk bg
    w = "full", // default value untuk w
  } = props;

  return (
    <button
      className={`flex items-center justify-center gap-1 ${p} py-1 ${bg} rounded-lg border-2 border-blue-500 font-medium w-${w} ${
        bg === "bg-blue-500" ? "text-white" : "text-blue-500"
      }`}
      onClick={onClick}
      disabled={disable}
    >
      <div className={`${hiddenL}`}>{logo}</div>
      <div className={`${hiddenT}`}>{title}</div>
    </button>
  );
};

export default Button;
