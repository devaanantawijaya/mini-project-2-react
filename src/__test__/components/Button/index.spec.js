import Button from "../../../components/Button";
import { render, fireEvent, screen } from "@testing-library/react";

describe("Button Component", () => {
  it("should render the button with the correct title", () => {
    render(<Button title="Click Me" />);
    expect(screen.getByText("Click Me")).toBeInTheDocument();
  });

  it("should call onClick function when clicked", () => {
    const handleClick = jest.fn();
    render(<Button title="Click Me" onClick={handleClick} />);
    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalled();
  });

  it("should be disabled when disable prop is true", () => {
    render(<Button title="Click Me" disable={true} />);
    const button = screen.getByRole("button");
    expect(button).toBeDisabled();
  });

  it("should apply custom class names based on props", () => {
    const { container } = render(
      <Button title="Custom" bg="bg-blue-500" w="full" />
    );
    const button = container.querySelector("button");
    screen.debug();

    expect(button.className).toContain("bg-blue-500");
    expect(button.className).toContain("w-full");
  });

  it("should match snapshot", () => {
    const { container } = render(<Button title="Snapshot Test" />);
    expect(container).toMatchSnapshot();
  });
});
