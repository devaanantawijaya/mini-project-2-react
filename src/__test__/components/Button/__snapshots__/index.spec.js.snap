// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapshot 1`] = `
<div>
  <button
    class="flex items-center justify-center gap-1 px-4 py-1 bg-blue-500 rounded-lg border-2 border-blue-500 font-medium w-full text-white"
  >
    <div
      class=""
    />
    <div
      class=""
    >
      Snapshot Test
    </div>
  </button>
</div>
`;
